builtin/bigint.mbt

```diff
@@ -290,110 +285,126 @@ fn grade_school_div(self : BigInt, other : BigInt) -> (BigInt, BigInt) {
   }
   if other.len == 1 {
     let number = other.limbs[0]
-    if number == 0 {
-      abort("divided by zero")
-    }
     let ret = self.deep_clone()
     if number == 1 {
       return (ret, zero)
     }
     let a = ret.limbs
-    let x = number.to_int64()
-    let mut y = 0L
+    let x = number.to_uint64()
+    let mut y = 0UL
     for i = self.len - 1; i >= 0; i = i - 1 {
       y = y.lsl(radix_bit_len)
-      y += a[i].to_int64()
-      a[i] = (y / x).land(radix_mask).to_int()
+      y += a[i].to_uint64()
+      a[i] = (y / x).land(radix_mask).to_uint()
       y %= x
     }
     if ret.limbs[ret.len - 1] == 0 {
       return (
         { ..ret, len: ret.len - 1 },
-        BigInt::from_int64(y.land(radix_mask)),
+        { limbs: FixedArray::make(1, y.to_uint()), sign: Positive, len: 1 },
       )
     }
-    return (ret, BigInt::from_int64(y.land(radix_mask)))
+    return (
+      ret,
+      { limbs: FixedArray::make(1, y.to_uint()), sign: Positive, len: 1 },
+    )
   }
 
   // Cite: TAOCP Vol. 2, 4.3.1
   let dividend = self
   let divisor = other
 
-  // normalize 
-  let mut lshift = 0
-  let mut back = divisor.limbs[divisor.len - 1].to_int64()
-  while back < radix / 2L {
-    back = back.lsl(1)
-    lshift += 1
-  }
+  // D1. normalize
+  // m = divident.limbs.length() - divisor.limbs.length()
+  // left shift divident & divisor such that 
+  // - divisor.limbs.last() >= radix / 2
+  // - divident.limbs.length() = self.limbs.length() + 1
```

Comment:
For test case `self = 18446744073709551615, other = 4294967296`, this condition doesn't hold, where `dividend.limbs.length() == 3, self.limbs.length() == 4`, and `a_len == 3`. In general, I think all invariants or pre/post conditions should not talk about `x.limbs.length()`, which is not part of the representation invariant (or at least it is not constrained by any upper bound).

Reply:
Yes, you are right. It should've been `len` instead of `limbs.length`